### ISSUE TITLE
Javascript "mixins" doesn't works if 'urlArgs' is in requirejs-config.js

### ISSUE THREAD

**Issue by azanelli** [magento/magento2#8221]
### Preconditions

1. Magento 2.1

### Steps to reproduce

- Create a new module `VendorName/MyModule`
- Add a `requirejs-config.js` file with this content:

```
// File: app/code/VendorName/MyModule/view/frontend/requirejs-config.js
var config = {
  'config':{
    'mixins': {
      'Magento_Checkout/js/view/shipping': {
         'VendorName_MyModule/hook':true
      }
    }
  },
  'urlArgs': "b=1"
};
```

- Create the `hook.js` file:

```
// File: app/code/VendorName/MyModule/view/base/web/hook.js
define([], function(){
  'use strict';    

  console.log("This is hook.js");

  return function(targetModule){
    return targetModule;
  };
});
```

- Go to the checkout page

### Expected result

-  `hook.js` should be loaded and we should se the string "This is hook.js" printed in the javascript's console

### Actual result

-  `hook.js` is **not** loaded and we do not see the string in the javascript's console

### Notes

- Note that if we remove the configuration `'urlArgs': "b=1"` from the `requirejs-config.js` file all it works fine and `hooks.js` will be loaded.

**Comment by veloraven** [comment]
@azanelli thank you for your report.
Please specify exact Magento version you are running.

**Comment by azanelli** [comment]
@veloraven Magento 2.1.2

**Comment by namnn** [comment]
@veloraven I have the same issue with a order comment module. I'm using Magento 2.1.0.

**Comment by jason-zz** [comment]
https://github.com/magento/magento2/blob/develop/lib/web/mage/requirejs/mixins.js#L116

URL augments should be removed in **path** variable. Otherwise, it won't match the key in requirejs-config.js. I think that's why mixins config gets fail when RequireJS defined **urlArgs**.

**Comment by thelettuce** [comment]
Thanks for the tip. Had the same problem when using urlArgs: "bust=2017-02-23" in Magento_Theme/templates/page/js/require_js.phtml

Fooman GoogleAnalyticsPlus wasn't working due to this problem.

My fix for lib/web/mage/requirejs/mixins.js is :
```
getMixins: function (path) {
	if (path.indexOf('?')!=-1) {
		path=path.substring(0,path.indexOf('?'));
	}
```
that seems to fix it.

**Comment by joachimVT** [comment]
I'm having the same issue on magento 2.1.6. Any updates on this ?

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-69674

**Comment by magento-engcom-team** [comment]
@azanelli, thank you for your report.
The issue is already fixed in 2.2.0
